management_app/views/auth.py

```diff
@@ -75,17 +76,29 @@ def login_callback():
     session["google_id"] = id_info.get("sub")
     session["name"] = id_info.get("name")
     session["domain"] = id_info.get("hd", "-1")
+    session['net_id'] = id_info.get('email').split(sep='@')[0]
 
     # verify the domain of email comes from UCI
     if session["domain"] != os.getenv("DOMAIN", ""):
         session.clear()
-        return redirect(url_for('auth.index', has_alert=True))
+        flash("Alert: You have to log in with your UCI email (e.g. abc123@uci.edu)")
+        return redirect(url_for('auth.index'))
 
+    # verify if the user has the permission to access the system
+    if not get_exist_user(session['net_id']):
+        session.clear()
+        flash("Alert: You do not have the permission to access the system.")
+        return redirect(url_for('auth.index'))
+
+    session['is_admin'] = check_admin(session['net_id'])
     return redirect(url_for('auth.redirect_to_homepage'))
 
 
 @auth.route("/redirect_to_homepage")
 @login_required
 def redirect_to_homepage():
-    # We currently use faculty page as home page after user logging in
-    return redirect('/faculty')
+    # We currently use faculty page as home page for admins after signing in
+    if session['is_admin']:
+        return redirect('/faculty')
+    else:
+        return redirect('/faculty')  # TODO: Redirect to regular user homepage
```

Comment:
Do we have any idea of what URL can be used to redirect the regular user to its home page?

Reply:
How about `/faculty-member`? We can keep this TODO and update it all together later when we implement the regular user page.